### DISCUSSION TITLE
1.40 - WSLg III: Works Even For Peasants edition

### DISCUSSION THREAD

**Discussion by cerebrate** [arkane-systems/genie#159] [General]
* Moved generic Linux/WSL functionality into a shared assembly.
 * Upnumbered genie-envar to fix missing path cloning on some systems.
 * Fixed typo in wslg-xwayland.socket (fixes #156).
 * Map XWayland socket only where WSLg is active and present.
 * Mount user runtime directory only where +WSLg (fixes #158) and where WSLg user matches (fixes #157).

<hr /><em>This discussion was created from the release <a href='https://github.com/arkane-systems/genie/releases/tag/v1.40'>1.40 - WSLg III: Works Even For Peasants edition</a>.</em>

**Comment by PavelSosin-320** [comment]
Greate! almost their. The Fedora distro starts almost completely after removing the old and installing the new version using rpm.
Except for one annoying unit: systemd-modules-load. WSL Kernel doesn't support dynamic kernel modules loading and no dynamically loadable modules are provided. This unit always fails. To avoid fake errors it can be masked using systemctl mask systemd-modules-load.

**Comment by PavelSosin-320** [comment]
Gently reminder: after login as aUser into the distro the overall control over sessions, users, and system state is handed over to the native  distro utilities like systemctl and loginctl. Thet can be invoked directly using wsl [-d distro] -u root -e loginctl  ... to gather information about users, sessions,etc, and managing users, sessions. Many services have both system-wide and user-level configurations. Be aware of your user and session id.
